server/src/metadata.rs

```diff
@@ -0,0 +1,1310 @@
+            .save_table_metadata("table_name")
+            .unwrap();
+
+        // Register the table with Apache Arrow DataFusion.
+        context.metadata_manager.register_tables(&context).unwrap();
+    }
+
+    #[test]
+    fn test_save_model_table_metadata() {
+        // Save a model table to the metadata database.
+        let temp_dir = tempfile::tempdir().unwrap();
+        let metadata_manager = test_util::get_test_metadata_manager(temp_dir.path());
+
+        let model_table_metadata = test_util::get_model_table_metadata();
+        metadata_manager
+            .save_model_table_metadata(&model_table_metadata, vec![1, 2, 4, 8])
+            .unwrap();
+
+        // Retrieve the model table from the metadata database.
+        let connection =
+            Connection::open(metadata_manager.metadata_database_path.to_str().unwrap()).unwrap();
+
+        // Check that an empty model_table_tags exists in the metadata database.
+        let mut select_statement = connection
+            .prepare("SELECT hash FROM model_table_tags")
+            .unwrap();
+        let mut rows = select_statement.query(()).unwrap();
+        assert!(rows.next().unwrap().is_none());
+
+        // Retrieve the rows in model_table_metadata from the metadata database.
+        let mut select_statement = connection
+            .prepare(
+                "SELECT table_name, schema, timestamp_column_index,
+                      tag_column_indices FROM model_table_metadata",
+            )
+            .unwrap();
+        let mut rows = select_statement.query(()).unwrap();
+
+        let row = rows.next().unwrap().unwrap();
+        assert_eq!("model_table", row.get::<usize, String>(0).unwrap());
+        assert_eq!(vec![1, 2, 4, 8], row.get::<usize, Vec<u8>>(1).unwrap());
+        assert_eq!(1, row.get::<usize, i32>(2).unwrap());
+        assert_eq!(vec![0], row.get::<usize, Vec<u8>>(3).unwrap());
+
+        assert!(rows.next().unwrap().is_none());
+
+        // Retrieve the rows in model_table_field_columns from the metadata
+        // database.
+        let mut select_statement = connection
+            .prepare(
+                "SELECT table_name, column_name, column_index
+                      FROM model_table_field_columns",
+            )
+            .unwrap();
+        let mut rows = select_statement.query(()).unwrap();
+
+        let row = rows.next().unwrap().unwrap();
+        assert_eq!("model_table", row.get::<usize, String>(0).unwrap());
+        assert_eq!("value_1", row.get::<usize, String>(1).unwrap());
+        assert_eq!(2, row.get::<usize, i32>(2).unwrap());
+
+        let row = rows.next().unwrap().unwrap();
+        assert_eq!("model_table", row.get::<usize, String>(0).unwrap());
+        assert_eq!("value_2", row.get::<usize, String>(1).unwrap());
+        assert_eq!(3, row.get::<usize, i32>(2).unwrap());
+
+        assert!(rows.next().unwrap().is_none());
+    }
+
+    #[test]
+    fn test_register_model_tables() {
```

Comment:
Is something missing in this test? it does not seem like anything is asserted.

Reply:
The test succeeds if none of the `unwrap()`s fail so no `assert!()`s are needed. A comment have been added to document this.